### PR TITLE
Ensure timestamp config makes sense.

### LINKED ISSUE #1493: Refactor timestamp.period to become double its value and called .expected_period
In Substrate 0.9 and 0.10 (0.10.0-78bb4c0) blocks are produced every 40 seconds when the genesis.runtime.timestamp.period is set to 20. Is this intended?

See "TSP antnet" in https://telemetry.polkadot.io/#/TSP%20antnet using the chainspec from https://github.com/webmaster128/tsp-networks/tree/master/chainspecs

### PR DESCRIPTION
**Pull request by gavofyork**
Closes #1493

### CODE REVIEW

**Comment by rphmeier** [comment]
That's a bit consensus-engine specific, isn't it?

File: srml/timestamp/src/lib.rs [hunk 1]
```diff
@@ -227,6 +227,15 @@ decl_module! {
 			<T::OnTimestampSet as OnTimestampSet<_>>::on_timestamp_set(now);
 		}
 
+		// Manage upgrade. Remove after all networks upgraded.
```
Comment (rphmeier) [inline srml/timestamp/src/lib.rs]: Can you link to https://github.com/paritytech/substrate/issues/2133 in the code? (our policy that TODOs all have issues, and this is basically a TODO)

**Comment by gavofyork** [comment]
Perhaps, but right now it's confusing to users and I don't see a better way of sorting it in time for 1.0.

**Comment by rphmeier** [comment]
We could add functions like `substrate_consensus_aura::minimum_period(block_time: Timestamp) -> Timestamp` and use those in the configuration

File: srml/timestamp/src/lib.rs [hunk 2]
```diff
@@ -211,14 +211,14 @@ decl_module! {
...
+		/// The timestamp should be greater than the previous one by the amount specified by `minimum_period`.
 		/// 
 		/// The dispatch origin for this call must be `Inherent`.
 		fn set(origin, #[compact] now: T::Moment) {
 			ensure_inherent(origin)?;
 			assert!(!<Self as Store>::DidUpdate::exists(), "Timestamp must be updated only once in the block");
 			assert!(
-				Self::now().is_zero() || now >= Self::now() + Self::block_period(),
+				Self::now().is_zero() || now >= Self::now() + <MinimumPeriod<T>>::get(),
 				"Timestamp must increment by at least <BlockPeriod> between sequential blocks"
```
Comment (shawntabrizi) [inline srml/timestamp/src/lib.rs]:
```suggestion
				"Timestamp must increment by at least <MinimumPeriod> between sequential blocks"
```

**Comment by gavofyork** [comment]
then genesis configs become pretty convoluted. and only to enable flexibility that 99.9% of users neither know nor care about

File: srml/timestamp/src/lib.rs [hunk 3]
```diff
@@ -368,10 +390,10 @@ mod tests {
 
 	#[test]
 	#[should_panic(expected = "Timestamp must increment by at least <BlockPeriod> between sequential blocks")]
```
Comment (shawntabrizi) [inline srml/timestamp/src/lib.rs]:
```suggestion
	#[should_panic(expected = "Timestamp must increment by at least <MinimumPeriod> between sequential blocks")]
```

**Comment by rphmeier** [comment]
> then genesis configs become pretty convoluted. and only to enable flexibility that 99.9% of users neither know nor care about

Sure, just worth noting we will almost definitely have to break compatibility with `timestamp` when rolling out `BABE`.

**Comment by gavofyork** [comment]
ok i'll replace with a straight rename to `minimum_period` and copious documentation. but it's still a pretty crappy dev experience and it'll need improving. the issue is clearly in the consensus module, which is where the block period should be configured and checked. timestamp should have the check removed entirely and just be a simple inherent + change-once-per-block-check + storage item.

**Pull request merged**